mlflow/cli/traces.py

```diff
@@ -0,0 +1,765 @@
+      data.spans.*                            # All span data
+      data.spans.*.name                       # Span operation names
+      data.spans.*.attributes.mlflow.spanType # Span types
+      data.spans.*.events.*.name              # Event names
+      info.trace_id,info.state,info.execution_duration  # Multiple fields
+    """
+
+
+@commands.command("search")
+@EXPERIMENT_ID
+@click.option(
+    "--filter-string",
+    type=click.STRING,
+    help="""Filter string for trace search.
+
+Examples:
+- Filter by run ID: "run_id = '123abc'"
+- Filter by status: "status = 'OK'"
+- Filter by timestamp: "timestamp_ms > 1700000000000"
+- Filter by metadata: "metadata.`mlflow.modelId` = 'model123'"
+- Filter by tags: "tags.environment = 'production'"
+- Multiple conditions: "run_id = '123' AND status = 'OK'"
+
+Available fields:
+- run_id: Associated MLflow run ID
+- status: Trace status (OK, ERROR, etc.)
+- timestamp_ms: Trace timestamp in milliseconds
+- execution_time_ms: Trace execution time in milliseconds
+- name: Trace name
+- metadata.<key>: Custom metadata fields (use backticks for keys with dots)
+- tags.<key>: Custom tag fields""",
+)
+@click.option(
+    "--max-results",
+    type=click.INT,
+    default=100,
+    help="Maximum number of traces to return (default: 100)",
+)
+@click.option(
+    "--order-by",
+    type=click.STRING,
+    help="Comma-separated list of fields to order by (e.g., 'timestamp_ms DESC, status')",
+)
+@click.option("--page-token", type=click.STRING, help="Token for pagination from previous search")
+@click.option(
+    "--run-id",
+    type=click.STRING,
+    help="Filter traces by run ID (convenience option, adds to filter-string)",
+)
+@click.option(
+    "--include-spans/--no-include-spans",
+    default=True,
+    help="Include span data in results (default: include)",
+)
+@click.option("--model-id", type=click.STRING, help="Filter traces by model ID")
+@click.option(
+    "--sql-warehouse-id",
+    type=click.STRING,
+    help="SQL warehouse ID for searching inference tables (Databricks only)",
+)
+@click.option(
+    "--output",
+    type=click.Choice(["table", "json"]),
+    default="table",
+    help="Output format: 'table' for formatted table (default) or 'json' for JSON format",
+)
+@click.option(
+    "--extract-fields",
+    type=click.STRING,
+    help="Filter and select specific fields using dot notation. "
+    'Examples: "info.trace_id", "info.assessments.*", "data.spans.*.name". '
+    'For field names with dots, use backticks: "info.tags.`mlflow.traceName`". '
+    "Comma-separated for multiple fields. "
+    "Defaults to standard columns for table mode, all fields for JSON mode.",
+)
+@click.option(
+    "--verbose",
+    is_flag=True,
+    help="Show all available fields in error messages when invalid fields are specified.",
+)
+def search_traces(
+    experiment_id: str,
+    filter_string: str,
+    max_results: int,
+    order_by: str,
+    page_token: str,
```

Comment:
```suggestion
    page_token: str | None,
```

is `None` passed when `--page-token` is unspecified?

Reply:
the same comment on other parameters.